src/go/agwd/config/config.go

```diff
@@ -53,24 +53,35 @@ const (
 
 // ParseTarget takes a target in string form and returns a resolved Target.
 // Extends functionality in grpc/internal/grpcutil.ParseTarget to support ipv4
-// and ipv6 schemes. Return tcp4Scheme or tcp6Scheme so the Scheme can be passed
-// directly to net.Listen.
+// and ipv6 schemes.
 func ParseTarget(target string) resolver.Target {
 	if strings.HasPrefix(target, ipv4Scheme+":") {
 		return resolver.Target{
-			Scheme:   tcp4Scheme,
+			Scheme:   ipv4Scheme,
 			Endpoint: target[len(ipv4Scheme)+1:],
 		}
 	}
 	if strings.HasPrefix(target, ipv6Scheme+":") {
 		return resolver.Target{
-			Scheme:   tcp6Scheme,
+			Scheme:   ipv6Scheme,
 			Endpoint: target[len(ipv6Scheme)+1:],
 		}
 	}
 	return grpcutil.ParseTarget(target, false)
 }
 
+// TargetSchemeAdapter takes a gRPC formatted Target such as {ipv4, ipv6}
+// and translates to net.Listen's required network type such as {tcp4, tcp6}.
+func TargetSchemeAdapter(target resolver.Target) resolver.Target {
```

Comment:
interesting. why not `func NewTargetListner(target resolver.Target) (net.Listner, error)`

Reply:
I started adding an injectable Factory interface for net.Listeners so I could write tests for it, but the code ended up being mostly abstractions to make it testable with almost no actual business logic. This was simpler to get under test coverage.